csrc/python_frontend/fusion_record.h

```diff
@@ -1264,7 +1264,19 @@ struct ConstantRecord : RecordFunctor {
     auto result = false;
     if (auto child_ptr = dynamic_cast<const ConstantRecord*>(&other)) {
       result = RecordFunctor::operator==(other);
-      result = result && (value_ == child_ptr->value_);
+      if (result) {
+        if constexpr (
+            std::is_same_v<ValueType, float> ||
+            std::is_same_v<ValueType, double>) {
+          if (std::isnan(value_) && std::isnan(child_ptr->value_)) {
```

Comment:
Can we foresee any problem with replacing one nan with another? On the one hand if someone is using signalling nan in a DL context is ill-advised. But we could also just convert to uint of appropriate size to compare. This also applies to `0 != -0` for signed ints and floats as well I guess..

Reply:
I guess the answer wrt nans in particular is that since Python doesn't distinguish nans and treats them all as quiet, we might as well also. For nans this makes sense because it will be printed in the kernel as a quiet `nan` even if the user supplied some different bits. Interestingly, somebody _could_ provide a `-0.0` which would print as such in the kernel (and be preserved in code as literal -0.0 is different from literal 0.0) but would test as equal in this function I believe.

Reply:
-0 and 0 can be inputed normally via python and don't require this special treatment as a string name.  So, I don't think we have an issue with them.